app/tree/gui/src/ts/twxplore/gui/tree/components/map/Map.tsx

```diff
@@ -2,128 +2,243 @@ import {connect, useDispatch, useSelector} from "react-redux";
+    const treeData =  getMapInfo({"variables" : {selectionInput: { includeBlocks: [blockUri], includeNtaList: [], excludeBlocks: [], excludeNtaList: [] }}})
+    dispatch(
+      addMapFeatures(
+        treeData.trees   data.boroughs.geometries.map(boroughFeature => ({
+          childType: MapFeatureType.NTA,
+          geometry: boroughFeature.geometry,
+          state: MapFeatureState.LOADED,
+          type: MapFeatureType.BOROUGH,
+          uri: boroughFeature.uri,
+        }))
+      )
+    );
+  }
+*/
+
+  if (state.features.length === 0) {
+    if (boroughsQueryResult.data) {
+      // Not tracking any features yet, add the boroughs we loaded
+      dispatch(
+        addMapFeatures(
+          boroughsQueryResult.data.boroughs.geometries.map(boroughFeature => ({
+            childType: MapFeatureType.NTA,
+            geometry: boroughFeature.geometry,
+            state: MapFeatureState.LOADED,
+            type: MapFeatureType.BOROUGH,
+            uri: boroughFeature.uri,
+          }))
+        )
+      );
+    }
+  }
+
+  // Organize the features by state
+  const featuresByState: {[index: string]: MapFeature[]} = {};
+  for (const feature of state.features) {
+    const features = featuresByState[feature.state];
+    if (features) {
+      features.push(feature);
+    } else {
+      featuresByState[feature.state] = [feature];
     }
+  }
 
-    // Feature state machine
-    for (const featureState in featuresByState) {
-        const featuresInState = featuresByState[featureState];
-        switch (featureState) {
-            case MapFeatureState.LOADED: {
-                const datasets = {
-                    data: Processors.processGeojson({
-                        "type": "FeatureCollection",
-                        "features": featuresInState.map(feature => {
-                            return {
-                                "type": "Feature",
-                                "geometry": wkt.parse(feature.geometry.wkt),
-                                "properties": feature
-                            }
-                        })
-                    }),
-                    info: {
-                        id: featuresInState[0].type
-                    }
-                }
-                dispatch(addDataToMap({datasets, options: {centerMap: true, readOnly: true}}));
-                break;
-            }
-            case MapFeatureState.CLICKED: {
-                for (const clickedFeature of featuresInState){
-                    getNtasByBorough({variables: {uri: clickedFeature.uri}});
-                    dispatch(changeMapFeatureState(clickedFeature.uri, MapFeatureState.RENDERED));
-                }
-                break;
-            }
-        
-        }        
+  // Feature state machine
+  for (const featureState in featuresByState) {
+    const featuresInState = featuresByState[featureState];
+    switch (featureState) {
+      case MapFeatureState.LOADED: {
+        const datasets = {
```

Comment:
This needs some more comments about what the state machine is doing. Feature has been LOADED (meaning what?), move it into X by dispatching ...